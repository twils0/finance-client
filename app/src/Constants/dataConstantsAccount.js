export const actionTypes = {
  SET_ACCOUNT_STATUS: 'DATA_ACCOUNT_SET_ACCOUNT_STATUS',
  SET_FIELD: 'DATA_ACCOUNT_SET_FIELD',
  SET_FIELDS: 'DATA_ACCOUNT_SET_FIELDS',
  RESET_ACCOUNT_STATE: 'DATA_ACCOUNT_RESET_ACCOUNT_STATE',
};

export const statusNames = {
  AWS_FIELDS: 'accountStatusAWSFields',
  STRIPE_FIELDS: 'accountStatusStripeFields',
  UPDATE_STRIPE_FIELDS_TOKEN: 'accountStatusUpdateStripeFieldsToken',
  UPDATE_STRIPE_FIELDS_REQUEST: 'accountStatusUpdateStripeFieldsRequest',
  UPDATE_DB_FIELDS: 'accountStatusUpdateDBFields',
  UPDATE_AWS_FIELDS: 'accountStatusUpdateAWSFields',
};

export const fieldNames = {
  NAME: 'accountName',
  EMAIL: 'accountEmail',
  EMAIL_ADDITIONAL: 'accountEmailAdditional',
  PHONE: 'accountPhone',
  NAME_ON_CARD: 'accountNameOnCard',
  PROMO_CODE: 'accountPromoCode',
  PROMO_CODE_VALID: 'accountPromoCodeValid',
  CUSTOMER: 'accountCustomer',
};
