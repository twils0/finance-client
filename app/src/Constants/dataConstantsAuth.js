export const actionTypes = {
  SET_AUTHENTICATED: 'DATA_AUTH_SET_AUTHENTICATED',
  SET_REDIRECT_URL: 'DATA_AUTH_SET_REDIRECT_URL',
  SET_AUTH_STATUS: 'DATA_AUTH_SET_AUTH_STATUS',
  SET_CODE_TYPE: 'DATA_AUTH_SET_CODE_TYPE',
  RESET_AUTH_STATE: 'DATA_AUTH_RESET_AUTH_STATE',
};

export const statusNames = {
  LOGIN: 'authStatusLogin',
  SIGN_OUT_DEVICES: 'authStatusSignOutDevices',
  LOGIN_MFA: 'authStatusLoginMFA',
  SIGN_UP: 'authStatusSignUp',
  VERIFY_PHONE: 'authStatusVerifyPhone',
  VERIFY_PHONE_CODE: 'authStatusVerifyPhoneCode',
  VERIFY_EMAIL: 'authStatusVerifyEmail',
  VERIFY_EMAIL_LINK: 'authStatusVerifyEmailLink',
  FORGOT_PASSWORD: 'authStatusForgotPassword',
  RESET_PASSWORD: 'authStatusResetPassword',
  CHANGE_PASSWORD: 'authStatusChangePassword',
  DELETE_ACCOUNT: 'authStatusDeleteAccount',
  LOGOUT: 'authStatusLogout',
};

export const codeTypeNames = {
  VERIFY_PHONE: 'verifyPhone',
  VERIFY_EMAIL: 'verifyEmail',
  VERIFY_EMAIL_ADDITIONAL: 'verifyEmailAdditional',
};
