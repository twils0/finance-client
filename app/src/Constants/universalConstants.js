import stripeConfig from '../../../stripe.config.json';
import dbConfig from '../../../db.config.json';
import iexConfig from '../../../iex.config.json';

export const baseURLs = {
  STRIPE: stripeConfig.url,
  DB: dbConfig.url,
  IEX: iexConfig.urls.api,
  IEX_DEV: iexConfig.urls.dev,
  IEX_TERMS: iexConfig.urls.terms,
};

export const URLs = {
  CUSTOMERS: '/customers',
  USERS: '/users',
  EMAILS: '/emails',
  SECURITIES: '/securities',
  SECURITIES_SEARCH: '/securities/search',
  IEX_COMPANY: tickerCusip => `/stock/${tickerCusip}/company`,
  IEX_QUOTE: tickerCusip => `/stock/${tickerCusip}/quote`,
};

export const axiosConfig = {
  STRIPE: {
    baseURL: baseURLs.STRIPE,
    timeout: 30000,
  },
  DB: {
    baseURL: baseURLs.DB,
    timeout: 30000,
  },
  IEX: {
    baseURL: baseURLs.IEX,
    timeout: 30000,
  },
};

export const pathNames = {
  DEFAULT: '*',
  LOGIN: '/login',
  SIGN_UP: '/signup',
  CODE: '/code',
  CODE_VERIFY_EMAIL: '/code/:verificationId?',
  TERMS: '/terms',
  INTERNAL: '/internal',
  WATCHLIST: '/internal/watchlist',
  WATCHLIST_SECURITY_ID: '/internal/watchlist/:securityId?',
  ACCOUNT: '/internal/account',
};

export const requestStatusTypes = {
  IDLE: 'IDLE',
  LOADING: 'LOADING',
  SUCCESS: 'SUCCESS',
  ERROR: 'ERROR',
};
