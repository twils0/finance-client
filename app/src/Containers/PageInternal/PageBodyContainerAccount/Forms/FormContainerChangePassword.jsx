import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import equals from 'validator/lib/equals';
import { BeatLoader } from 'halogenium';

import theme from '../../../../themes';

import { requestStatusTypes } from '../../../../Constants/universalConstants';
import { statusNames } from '../../../../Constants/dataConstantsAuth';
import { errorMessages } from '../../../../Constants/uiConstantsApp';
import { formNames, inputNames } from '../../../../Constants/uiConstantsAccount';
import { setInputValueError } from '../../../../Actions/uiActionsAccount';

import FlexColumn from '../../../../Components/Flex/FlexColumn';
import FormChangePassword from '../../../../Components/PageInternal/PageBodyAccount/Forms/FormChangePassword';

class FormContainerChangePassword extends React.Component {
  componentDidMount() {
    this.props.setHeight(formNames.CHANGE_PASSWORD);
  }

  componentWillUpdate(nextProps) {
    if (nextProps.statusAuth[statusNames.CHANGE_PASSWORD].status === requestStatusTypes.LOADING) {
      nextProps.clearElement(formNames.CHANGE_PASSWORD);
    }
  }

  componentDidUpdate() {
    this.props.setHeight(formNames.CHANGE_PASSWORD);
  }

  componentWillUnmount() {
    this.props.clearElement(formNames.CHANGE_PASSWORD);
  }

  handleInputChange = (inputEvent) => {
    inputEvent.preventDefault();
    const { inputs, handleInputValueError } = this.props;
    const input = { ...inputs[inputEvent.target.id] };

    if (input) {
      if (inputEvent.target.value || input.value) {
        input.value = inputEvent.target.value;
      }

      if (input.errorMessage) {
        switch (input.errorMessage) {
          case errorMessages.NO_PASSWORD: {
            if (input.value) {
              input.errorMessage = '';
            }
            break;
          }
          case errorMessages.NO_NEW_PASSWORD: {
            if (input.value) {
              input.errorMessage = '';
            }
            break;
          }
          case errorMessages.INVALID_PASSWORD: {
            const passwordRegex = /^(?=.*[A-Za-z])(?=.*\d)(?=.*[$@$!%*#?&])[A-Za-z\d$@$!%*#?&]{8,}$/;
            if (!passwordRegex.test(input.value)) {
              input.errorMessage = '';
            }
            break;
          }
          case errorMessages.NO_MATCH_PASSWORD: {
            if (
              equals(
                inputs[inputNames[formNames.CHANGE_PASSWORD].NEW_PASSWORD].value,
                inputs[inputNames[formNames.CHANGE_PASSWORD].NEW_PASSWORD2].value,
              )
            ) {
              input.errorMessage = '';
            }
            break;
          }
          case errorMessages.PASSWORD_NOT_FOUND: {
            if (input.value) {
              input.errorMessage = '';
            }
            break;
          }
          default:
            input.errorMessage = '';
            break;
        }
      }
    }

    handleInputValueError(input);
  };

  render() {
    const {
      heightRef,
      rowWidthIndent,
      headerWidthBufferRight,
      mediaWidth,
      height,
      rowHeight,
      rowHeightBuffer,
      statusAuth,
      currentForm,
      inputs,
    } = this.props;

    if (statusAuth[statusNames.CHANGE_PASSWORD].status === requestStatusTypes.LOADING) {
      const heightString = `${height}px`;

      return (
        <FlexColumn width="100%" height={heightString}>
          <BeatLoader color={theme.color.secondary.default} />
        </FlexColumn>
      );
    }
    return (
      <FormChangePassword
        heightRef={heightRef}
        rowWidthIndent={rowWidthIndent}
        headerWidthBufferRight={headerWidthBufferRight}
        mediaWidth={mediaWidth}
        rowHeight={rowHeight}
        rowHeightBuffer={rowHeightBuffer}
        currentFormBool={currentForm === formNames.CHANGE_PASSWORD}
        inputs={inputs}
        handleInputChange={this.handleInputChange}
      />
    );
  }
}

FormContainerChangePassword.propTypes = {
  heightRef: PropTypes.func.isRequired,
  rowWidthIndent: PropTypes.number.isRequired,
  headerWidthBufferRight: PropTypes.number.isRequired,
  mediaWidth: PropTypes.number.isRequired,
  height: PropTypes.number.isRequired,
  rowHeight: PropTypes.number.isRequired,
  rowHeightBuffer: PropTypes.number.isRequired,
  statusAuth: PropTypes.object.isRequired,
  currentForm: PropTypes.string.isRequired,
  inputs: PropTypes.object.isRequired,
  setHeight: PropTypes.func.isRequired,
  clearElement: PropTypes.func.isRequired,
  handleInputValueError: PropTypes.func.isRequired,
};

FormContainerChangePassword.defaultProps = {};

const mapStateToProps = (state, ownProps) => ({
  heightRef: ownProps.heightRef,
  rowWidthIndent: state.ui.internal.account.utilities.forms.rowWidthIndent,
  headerWidthBufferRight: state.ui.internal.account.utilities.forms.headerWidthBufferRight,
  mediaWidth: state.ui.internal.account.utilities.pageBody.mediaWidth,
  height: state.ui.internal.account.forms[formNames.BILLING].height,
  rowHeight: state.ui.internal.account.utilities.forms.rowHeight,
  rowHeightBuffer: state.ui.internal.account.utilities.forms.rowHeightBuffer,
  heightBuffer: state.ui.internal.account.utilities.forms.heightBuffer,
  statusAuth: state.data.auth.status,
  currentForm: state.ui.internal.account.forms.current,
  inputs: state.ui.internal.account.forms[formNames.CHANGE_PASSWORD].inputs,
  setHeight: ownProps.setHeight,
  clearElement: ownProps.clearElement,
});

const mapDispatchToProps = dispatch => ({
  handleInputValueError: payload => dispatch(setInputValueError(payload)),
});

export default connect(mapStateToProps, mapDispatchToProps)(FormContainerChangePassword);
