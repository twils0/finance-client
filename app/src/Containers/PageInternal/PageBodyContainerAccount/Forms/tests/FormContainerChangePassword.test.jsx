import React from 'react';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import { shallow } from 'enzyme';

import { initialState as initialStateAuth } from '../../../../../Reducers/dataReducersAuth';
import { initialState as initialStateUIAccount } from '../../../../../Reducers/uiReducersAccount';
import { requestStatusTypes } from '../../../../../Constants/universalConstants';
import { statusNames } from '../../../../../Constants/dataConstantsAuth';
import { errorMessages } from '../../../../../Constants/uiConstantsApp';
import {
  actionTypes as actionTypesUIAccount,
  formNames,
  inputNames,
} from '../../../../../Constants/uiConstantsAccount';

import FormContainerChangePassword from '../FormContainerChangePassword';

const middleware = [thunk];
const mockStore = configureMockStore(middleware);

const heightRef = jest.fn();
const setHeight = jest.fn();
const clearElement = jest.fn();

const oldPassword = 'testOldPassword1!';
const password = 'testPassword1!';

const shallowComponent = store => shallow(
  <FormContainerChangePassword
    heightRef={heightRef}
    setHeight={setHeight}
    clearElement={clearElement}
  />,
  {
    context: {
      store,
    },
  },
).dive();

describe('Containers', () => {
  describe('PageInternal', () => {
    describe('PageBodyContainerAccount', () => {
      describe('Forms', () => {
        describe('FormContainerChangePassword', () => {
          afterEach(() => {
            heightRef.mockReset();
            setHeight.mockReset();
            clearElement.mockReset();
          });

          it('shallow renders correctly', async () => {
            const stateBeforeUIAccount = JSON.parse(JSON.stringify(initialStateUIAccount));

            const store = mockStore({
              data: {
                auth: initialStateAuth,
              },
              ui: {
                internal: {
                  account: stateBeforeUIAccount,
                },
              },
            });

            const expectedActions = [];

            const wrapper = shallowComponent(store);

            const actions = store.getActions();

            expect(actions).toEqual(expectedActions);
            expect(wrapper).toMatchSnapshot();
            expect(setHeight).toBeCalledWith(formNames.CHANGE_PASSWORD);
          });

          it('shallow renders correctly, with componentWillUpdate, componentDidUpdate, and componentWillUnmount', async () => {
            const stateBeforeUIAccount = JSON.parse(JSON.stringify(initialStateUIAccount));

            const store = mockStore({
              data: {
                auth: initialStateAuth,
              },
              ui: {
                internal: {
                  account: stateBeforeUIAccount,
                },
              },
            });

            const nextProps = {
              statusAuth: {
                [statusNames.CHANGE_PASSWORD]: {
                  status: requestStatusTypes.LOADING,
                },
              },
              clearElement,
            };

            const expectedActions = [];

            const wrapper = shallowComponent(store);

            const instance = wrapper.instance();

            instance.componentWillUpdate(nextProps);
            instance.componentDidUpdate();
            instance.componentWillUnmount();

            const actions = store.getActions();

            expect(actions).toEqual(expectedActions);
            expect(wrapper).toMatchSnapshot();
            expect(setHeight).toBeCalledWith(formNames.CHANGE_PASSWORD);
            expect(clearElement).toBeCalledWith(formNames.CHANGE_PASSWORD);
            expect(setHeight).toBeCalledWith(formNames.CHANGE_PASSWORD);
            expect(clearElement).toBeCalledWith(formNames.CHANGE_PASSWORD);
            expect(setHeight).toHaveBeenCalledTimes(2);
            expect(clearElement).toHaveBeenCalledTimes(2);
          });

          it('shallow renders correctly, with loading indicator', async () => {
            const stateBeforeAuth = JSON.parse(JSON.stringify(initialStateAuth));
            const stateBeforeUIAccount = JSON.parse(JSON.stringify(initialStateUIAccount));

            stateBeforeAuth.status[statusNames.CHANGE_PASSWORD].status = requestStatusTypes.LOADING;

            const store = mockStore({
              data: {
                auth: stateBeforeAuth,
              },
              ui: {
                internal: {
                  account: stateBeforeUIAccount,
                },
              },
            });

            const expectedActions = [];

            const wrapper = shallowComponent(store);

            const actions = store.getActions();

            expect(actions).toEqual(expectedActions);
            expect(wrapper).toMatchSnapshot();
          });

          it('shallow renders correctly and calls handleInputChange, with no password error', async () => {
            const stateBeforeUIAccount = JSON.parse(JSON.stringify(initialStateUIAccount));
            const { forms } = stateBeforeUIAccount;

            forms[formNames.CHANGE_PASSWORD].inputs[
              inputNames[formNames.CHANGE_PASSWORD].OLD_PASSWORD
            ].errorMessage = errorMessages.NO_PASSWORD;

            const store = mockStore({
              data: {
                auth: initialStateAuth,
              },
              ui: {
                internal: {
                  account: stateBeforeUIAccount,
                },
              },
            });

            const uiOldPassword = JSON.parse(
              JSON.stringify(
                forms[formNames.CHANGE_PASSWORD].inputs[
                  inputNames[formNames.CHANGE_PASSWORD].OLD_PASSWORD
                ],
              ),
            );

            uiOldPassword.value = oldPassword;
            uiOldPassword.errorMessage = '';

            const inputEvent = {
              preventDefault: jest.fn(),
              target: {
                id: inputNames[formNames.CHANGE_PASSWORD].OLD_PASSWORD,
                value: oldPassword,
              },
            };
            const expectedActions = [
              {
                type: actionTypesUIAccount.SET_INPUT_VALUE_ERROR,
                payload: uiOldPassword,
              },
            ];

            const wrapper = shallowComponent(store);

            const instance = wrapper.instance();

            instance.handleInputChange(inputEvent);

            const actions = store.getActions();

            expect(actions).toEqual(expectedActions);
            expect(wrapper).toMatchSnapshot();
          });

          it('shallow renders correctly and calls handleInputChange, with no new password error', async () => {
            const stateBeforeUIAccount = JSON.parse(JSON.stringify(initialStateUIAccount));
            const { forms } = stateBeforeUIAccount;

            forms[formNames.CHANGE_PASSWORD].inputs[
              inputNames[formNames.CHANGE_PASSWORD].NEW_PASSWORD
            ].errorMessage = errorMessages.NO_NEW_PASSWORD;

            const store = mockStore({
              data: {
                auth: initialStateAuth,
              },
              ui: {
                internal: {
                  account: stateBeforeUIAccount,
                },
              },
            });

            const uiNewPassword = JSON.parse(
              JSON.stringify(
                forms[formNames.CHANGE_PASSWORD].inputs[
                  inputNames[formNames.CHANGE_PASSWORD].NEW_PASSWORD
                ],
              ),
            );

            uiNewPassword.value = password;
            uiNewPassword.errorMessage = '';

            const inputEvent = {
              preventDefault: jest.fn(),
              target: {
                id: inputNames[formNames.CHANGE_PASSWORD].NEW_PASSWORD,
                value: password,
              },
            };
            const expectedActions = [
              {
                type: actionTypesUIAccount.SET_INPUT_VALUE_ERROR,
                payload: uiNewPassword,
              },
            ];

            const wrapper = shallowComponent(store);

            const instance = wrapper.instance();

            instance.handleInputChange(inputEvent);

            const actions = store.getActions();

            expect(actions).toEqual(expectedActions);
            expect(wrapper).toMatchSnapshot();
          });

          it('shallow renders correctly and calls handleInputChange, with invalid password error', async () => {
            const stateBeforeUIAccount = JSON.parse(JSON.stringify(initialStateUIAccount));
            const { forms } = stateBeforeUIAccount;

            const wrongPassword = 'test';

            forms[formNames.CHANGE_PASSWORD].inputs[
              inputNames[formNames.CHANGE_PASSWORD].NEW_PASSWORD
            ].value = wrongPassword;
            forms[formNames.CHANGE_PASSWORD].inputs[
              inputNames[formNames.CHANGE_PASSWORD].NEW_PASSWORD
            ].errorMessage = errorMessages.INVALID_PASSWORD;

            const store = mockStore({
              data: {
                auth: initialStateAuth,
              },
              ui: {
                internal: {
                  account: stateBeforeUIAccount,
                },
              },
            });

            const uiNewPassword = JSON.parse(
              JSON.stringify(
                forms[formNames.CHANGE_PASSWORD].inputs[
                  inputNames[formNames.CHANGE_PASSWORD].NEW_PASSWORD
                ],
              ),
            );

            uiNewPassword.value = password;

            const inputEvent = {
              preventDefault: jest.fn(),
              target: {
                id: inputNames[formNames.CHANGE_PASSWORD].NEW_PASSWORD,
                value: password,
              },
            };
            const expectedActions = [
              {
                type: actionTypesUIAccount.SET_INPUT_VALUE_ERROR,
                payload: uiNewPassword,
              },
            ];

            const wrapper = shallowComponent(store);

            const instance = wrapper.instance();

            instance.handleInputChange(inputEvent);

            const actions = store.getActions();

            expect(actions).toEqual(expectedActions);
            expect(wrapper).toMatchSnapshot();
          });

          it('shallow renders correctly and calls handleInputChange, with no match password', async () => {
            const stateBeforeUIAccount = JSON.parse(JSON.stringify(initialStateUIAccount));
            const { forms } = stateBeforeUIAccount;

            forms[formNames.CHANGE_PASSWORD].inputs[
              inputNames[formNames.CHANGE_PASSWORD].NEW_PASSWORD
            ].value = password;
            forms[formNames.CHANGE_PASSWORD].inputs[
              inputNames[formNames.CHANGE_PASSWORD].NEW_PASSWORD2
            ].value = password;
            forms[formNames.CHANGE_PASSWORD].inputs[
              inputNames[formNames.CHANGE_PASSWORD].NEW_PASSWORD2
            ].errorMessage = errorMessages.NO_MATCH_PASSWORD;

            const store = mockStore({
              data: {
                auth: initialStateAuth,
              },
              ui: {
                internal: {
                  account: stateBeforeUIAccount,
                },
              },
            });

            const uiNewPassword2 = JSON.parse(
              JSON.stringify(
                forms[formNames.CHANGE_PASSWORD].inputs[
                  inputNames[formNames.CHANGE_PASSWORD].NEW_PASSWORD2
                ],
              ),
            );

            uiNewPassword2.value = password;
            uiNewPassword2.errorMessage = '';

            const inputEvent = {
              preventDefault: jest.fn(),
              target: {
                id: inputNames[formNames.CHANGE_PASSWORD].NEW_PASSWORD2,
                value: password,
              },
            };
            const expectedActions = [
              {
                type: actionTypesUIAccount.SET_INPUT_VALUE_ERROR,
                payload: uiNewPassword2,
              },
            ];

            const wrapper = shallowComponent(store);

            const instance = wrapper.instance();

            instance.handleInputChange(inputEvent);

            const actions = store.getActions();

            expect(actions).toEqual(expectedActions);
            expect(wrapper).toMatchSnapshot();
          });

          it('shallow renders correctly and calls handleInputChange, with password not found', async () => {
            const stateBeforeUIAccount = JSON.parse(JSON.stringify(initialStateUIAccount));
            const { forms } = stateBeforeUIAccount;

            const wrongPassword = 'test';

            forms[formNames.CHANGE_PASSWORD].inputs[
              inputNames[formNames.CHANGE_PASSWORD].NEW_PASSWORD
            ].value = wrongPassword;
            forms[formNames.CHANGE_PASSWORD].inputs[
              inputNames[formNames.CHANGE_PASSWORD].NEW_PASSWORD
            ].errorMessage = errorMessages.PASSWORD_NOT_FOUND;

            const store = mockStore({
              data: {
                auth: initialStateAuth,
              },
              ui: {
                internal: {
                  account: stateBeforeUIAccount,
                },
              },
            });

            const uiNewPassword = JSON.parse(
              JSON.stringify(
                forms[formNames.CHANGE_PASSWORD].inputs[
                  inputNames[formNames.CHANGE_PASSWORD].NEW_PASSWORD
                ],
              ),
            );

            uiNewPassword.value = password;
            uiNewPassword.errorMessage = '';

            const inputEvent = {
              preventDefault: jest.fn(),
              target: {
                id: inputNames[formNames.CHANGE_PASSWORD].NEW_PASSWORD,
                value: password,
              },
            };
            const expectedActions = [
              {
                type: actionTypesUIAccount.SET_INPUT_VALUE_ERROR,
                payload: uiNewPassword,
              },
            ];

            const wrapper = shallowComponent(store);

            const instance = wrapper.instance();

            instance.handleInputChange(inputEvent);

            const actions = store.getActions();

            expect(actions).toEqual(expectedActions);
            expect(wrapper).toMatchSnapshot();
          });

          it('shallow renders correctly and calls handleInputChange, with undefined error', async () => {
            const stateBeforeUIAccount = JSON.parse(JSON.stringify(initialStateUIAccount));
            const { forms } = stateBeforeUIAccount;

            forms[formNames.CHANGE_PASSWORD].inputs[
              inputNames[formNames.CHANGE_PASSWORD].NEW_PASSWORD2
            ].errorMessage = errorMessages.INTERNAL_ERROR;

            const store = mockStore({
              data: {
                auth: initialStateAuth,
              },
              ui: {
                internal: {
                  account: stateBeforeUIAccount,
                },
              },
            });

            const uiNewPassword2 = JSON.parse(
              JSON.stringify(
                forms[formNames.CHANGE_PASSWORD].inputs[
                  inputNames[formNames.CHANGE_PASSWORD].NEW_PASSWORD2
                ],
              ),
            );

            uiNewPassword2.value = password;
            uiNewPassword2.errorMessage = '';

            const inputEvent = {
              preventDefault: jest.fn(),
              target: {
                id: inputNames[formNames.CHANGE_PASSWORD].NEW_PASSWORD2,
                value: password,
              },
            };
            const expectedActions = [
              {
                type: actionTypesUIAccount.SET_INPUT_VALUE_ERROR,
                payload: uiNewPassword2,
              },
            ];

            const wrapper = shallowComponent(store);

            const instance = wrapper.instance();

            instance.handleInputChange(inputEvent);

            const actions = store.getActions();

            expect(actions).toEqual(expectedActions);
            expect(wrapper).toMatchSnapshot();
          });
        });
      });
    });
  });
});
