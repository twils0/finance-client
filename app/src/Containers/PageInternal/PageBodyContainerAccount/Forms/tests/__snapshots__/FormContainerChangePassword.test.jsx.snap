// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Containers PageInternal PageBodyContainerAccount Forms FormContainerChangePassword shallow renders correctly 1`] = `
<FormChangePassword
  currentFormBool={false}
  handleInputChange={[Function]}
  headerWidthBufferRight={10}
  heightRef={[MockFunction]}
  inputs={
    Object {
      "changePasswordNewPassword": Object {
        "errorMessage": "",
        "id": "changePasswordNewPassword",
        "parentId": "changePasswordForm",
        "value": "",
      },
      "changePasswordNewPassword2": Object {
        "errorMessage": "",
        "id": "changePasswordNewPassword2",
        "parentId": "changePasswordForm",
        "value": "",
      },
      "changePasswordOldPassword": Object {
        "errorMessage": "",
        "id": "changePasswordOldPassword",
        "parentId": "changePasswordForm",
        "value": "",
      },
    }
  }
  mediaWidth={874}
  rowHeight={60}
  rowHeightBuffer={18}
  rowWidthIndent={10}
/>
`;

exports[`Containers PageInternal PageBodyContainerAccount Forms FormContainerChangePassword shallow renders correctly and calls handleInputChange, with invalid password error 1`] = `
<FormChangePassword
  currentFormBool={false}
  handleInputChange={[Function]}
  headerWidthBufferRight={10}
  heightRef={[MockFunction]}
  inputs={
    Object {
      "changePasswordNewPassword": Object {
        "errorMessage": "Please enter a password containing at least 8 characters, one of the following special characters, $@!%*#?&, one letter, and one number.",
        "id": "changePasswordNewPassword",
        "parentId": "changePasswordForm",
        "value": "test",
      },
      "changePasswordNewPassword2": Object {
        "errorMessage": "",
        "id": "changePasswordNewPassword2",
        "parentId": "changePasswordForm",
        "value": "",
      },
      "changePasswordOldPassword": Object {
        "errorMessage": "",
        "id": "changePasswordOldPassword",
        "parentId": "changePasswordForm",
        "value": "",
      },
    }
  }
  mediaWidth={874}
  rowHeight={60}
  rowHeightBuffer={18}
  rowWidthIndent={10}
/>
`;

exports[`Containers PageInternal PageBodyContainerAccount Forms FormContainerChangePassword shallow renders correctly and calls handleInputChange, with no match password 1`] = `
<FormChangePassword
  currentFormBool={false}
  handleInputChange={[Function]}
  headerWidthBufferRight={10}
  heightRef={[MockFunction]}
  inputs={
    Object {
      "changePasswordNewPassword": Object {
        "errorMessage": "",
        "id": "changePasswordNewPassword",
        "parentId": "changePasswordForm",
        "value": "testPassword1!",
      },
      "changePasswordNewPassword2": Object {
        "errorMessage": "This password does not match your password above.",
        "id": "changePasswordNewPassword2",
        "parentId": "changePasswordForm",
        "value": "testPassword1!",
      },
      "changePasswordOldPassword": Object {
        "errorMessage": "",
        "id": "changePasswordOldPassword",
        "parentId": "changePasswordForm",
        "value": "",
      },
    }
  }
  mediaWidth={874}
  rowHeight={60}
  rowHeightBuffer={18}
  rowWidthIndent={10}
/>
`;

exports[`Containers PageInternal PageBodyContainerAccount Forms FormContainerChangePassword shallow renders correctly and calls handleInputChange, with no new password error 1`] = `
<FormChangePassword
  currentFormBool={false}
  handleInputChange={[Function]}
  headerWidthBufferRight={10}
  heightRef={[MockFunction]}
  inputs={
    Object {
      "changePasswordNewPassword": Object {
        "errorMessage": "Please enter your new password.",
        "id": "changePasswordNewPassword",
        "parentId": "changePasswordForm",
        "value": "",
      },
      "changePasswordNewPassword2": Object {
        "errorMessage": "",
        "id": "changePasswordNewPassword2",
        "parentId": "changePasswordForm",
        "value": "",
      },
      "changePasswordOldPassword": Object {
        "errorMessage": "",
        "id": "changePasswordOldPassword",
        "parentId": "changePasswordForm",
        "value": "",
      },
    }
  }
  mediaWidth={874}
  rowHeight={60}
  rowHeightBuffer={18}
  rowWidthIndent={10}
/>
`;

exports[`Containers PageInternal PageBodyContainerAccount Forms FormContainerChangePassword shallow renders correctly and calls handleInputChange, with no password error 1`] = `
<FormChangePassword
  currentFormBool={false}
  handleInputChange={[Function]}
  headerWidthBufferRight={10}
  heightRef={[MockFunction]}
  inputs={
    Object {
      "changePasswordNewPassword": Object {
        "errorMessage": "",
        "id": "changePasswordNewPassword",
        "parentId": "changePasswordForm",
        "value": "",
      },
      "changePasswordNewPassword2": Object {
        "errorMessage": "",
        "id": "changePasswordNewPassword2",
        "parentId": "changePasswordForm",
        "value": "",
      },
      "changePasswordOldPassword": Object {
        "errorMessage": "Please enter your password.",
        "id": "changePasswordOldPassword",
        "parentId": "changePasswordForm",
        "value": "",
      },
    }
  }
  mediaWidth={874}
  rowHeight={60}
  rowHeightBuffer={18}
  rowWidthIndent={10}
/>
`;

exports[`Containers PageInternal PageBodyContainerAccount Forms FormContainerChangePassword shallow renders correctly and calls handleInputChange, with password not found 1`] = `
<FormChangePassword
  currentFormBool={false}
  handleInputChange={[Function]}
  headerWidthBufferRight={10}
  heightRef={[MockFunction]}
  inputs={
    Object {
      "changePasswordNewPassword": Object {
        "errorMessage": "This password does not match our records.",
        "id": "changePasswordNewPassword",
        "parentId": "changePasswordForm",
        "value": "test",
      },
      "changePasswordNewPassword2": Object {
        "errorMessage": "",
        "id": "changePasswordNewPassword2",
        "parentId": "changePasswordForm",
        "value": "",
      },
      "changePasswordOldPassword": Object {
        "errorMessage": "",
        "id": "changePasswordOldPassword",
        "parentId": "changePasswordForm",
        "value": "",
      },
    }
  }
  mediaWidth={874}
  rowHeight={60}
  rowHeightBuffer={18}
  rowWidthIndent={10}
/>
`;

exports[`Containers PageInternal PageBodyContainerAccount Forms FormContainerChangePassword shallow renders correctly and calls handleInputChange, with undefined error 1`] = `
<FormChangePassword
  currentFormBool={false}
  handleInputChange={[Function]}
  headerWidthBufferRight={10}
  heightRef={[MockFunction]}
  inputs={
    Object {
      "changePasswordNewPassword": Object {
        "errorMessage": "",
        "id": "changePasswordNewPassword",
        "parentId": "changePasswordForm",
        "value": "",
      },
      "changePasswordNewPassword2": Object {
        "errorMessage": "Please try your request again. If unsuccessful on subsequent attempts, please contact customer service.",
        "id": "changePasswordNewPassword2",
        "parentId": "changePasswordForm",
        "value": "",
      },
      "changePasswordOldPassword": Object {
        "errorMessage": "",
        "id": "changePasswordOldPassword",
        "parentId": "changePasswordForm",
        "value": "",
      },
    }
  }
  mediaWidth={874}
  rowHeight={60}
  rowHeightBuffer={18}
  rowWidthIndent={10}
/>
`;

exports[`Containers PageInternal PageBodyContainerAccount Forms FormContainerChangePassword shallow renders correctly, with componentWillUpdate, componentDidUpdate, and componentWillUnmount 1`] = `
<FormChangePassword
  currentFormBool={false}
  handleInputChange={[Function]}
  headerWidthBufferRight={10}
  heightRef={[MockFunction]}
  inputs={
    Object {
      "changePasswordNewPassword": Object {
        "errorMessage": "",
        "id": "changePasswordNewPassword",
        "parentId": "changePasswordForm",
        "value": "",
      },
      "changePasswordNewPassword2": Object {
        "errorMessage": "",
        "id": "changePasswordNewPassword2",
        "parentId": "changePasswordForm",
        "value": "",
      },
      "changePasswordOldPassword": Object {
        "errorMessage": "",
        "id": "changePasswordOldPassword",
        "parentId": "changePasswordForm",
        "value": "",
      },
    }
  }
  mediaWidth={874}
  rowHeight={60}
  rowHeightBuffer={18}
  rowWidthIndent={10}
/>
`;

exports[`Containers PageInternal PageBodyContainerAccount Forms FormContainerChangePassword shallow renders correctly, with loading indicator 1`] = `
<FlexColumn
  alignItems={null}
  alignSelf={null}
  borderColorType="border"
  borderRadius={null}
  borderStyle={null}
  borderWidth={null}
  boxSizing={null}
  color={Object {}}
  display={null}
  flex={null}
  flexWrap={null}
  height="0px"
  justifyContent={null}
  margin={null}
  mediaWidth={null}
  minHeight={null}
  minWidth={null}
  overflowX={null}
  overflowY={null}
  padding={null}
  position={null}
  responsiveAlignSecurities={null}
  responsiveJustifyContent={null}
  responsiveMargin={null}
  responsivePadding={null}
  responsivePosition={null}
  responsiveWidth={null}
  separator={false}
  themeColor="primary"
  width="100%"
>
  <BeatLoader
    color="#7367e6"
    loading={true}
    margin="2px"
    size="15px"
  />
</FlexColumn>
`;
